packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.test.tsx

```diff
@@ -16,32 +16,95 @@ const account = {
   ...MOCK_ACCOUNT,
 } as unknown as Account;
 
+const accountWithLongEmail = {
+  ...MOCK_ACCOUNT,
+  primaryEmail: {
+    email:
+      'supercalifragilisticexpialidocious@marypoppins.superfan.conference.com',
+  },
+} as unknown as Account;
+
 jest.mock('../../lib/metrics', () => ({
   logViewEvent: jest.fn(),
 }));
 
-it('renders button as expected', () => {
+beforeAll(() => {
   window.URL.createObjectURL = jest.fn();
-  render(
-    <AppContext.Provider value={{ account }}>
-      <ButtonDownloadRecoveryKey {...{ recoveryKeyValue, viewName }} />
-    </AppContext.Provider>
-  );
-  expect(screen.getByText('Download your recovery key')).toBeInTheDocument();
 });
 
-it('emits a metrics event when the link is clicked', async () => {
-  window.URL.createObjectURL = jest.fn();
-  render(
-    <AppContext.Provider value={{ account }}>
-      <ButtonDownloadRecoveryKey {...{ recoveryKeyValue, viewName }} />
-    </AppContext.Provider>
-  );
-  const downloadButton = screen.getByText('Download your recovery key');
-  fireEvent.click(downloadButton);
-
-  expect(logViewEvent).toHaveBeenCalledWith(
-    `flow.${viewName}`,
-    'recovery-key.download-option'
-  );
+describe('ButtonDownloadRecoveryKey', () => {
+  it('renders button as expected', () => {
+    render(
+      <AppContext.Provider value={{ account }}>
+        <ButtonDownloadRecoveryKey {...{ recoveryKeyValue, viewName }} />
+      </AppContext.Provider>
+    );
+    screen.getByText('Download your account recovery key');
+  });
+
+  it('sets the filename as expected with a reasonably-sized email', () => {
+    render(
+      <AppContext.Provider value={{ account }}>
+        <ButtonDownloadRecoveryKey {...{ recoveryKeyValue, viewName }} />
+      </AppContext.Provider>
+    );
+    const downloadButtonDownloadAttribute = screen
+      .getByText('Download your account recovery key')
+      .getAttribute('download');
+
+    // Test the date formatting
+    const mockDateObject = '2023-05-10T17:00:40.722Z';
+    const mockDateFunctionOutput = new Date(mockDateObject)
+      .toISOString()
+      .split('T')[0];
+    const mockExpectedDate = '2023-05-10';
+
+    expect(mockDateFunctionOutput).toEqual(mockExpectedDate);
+
+    // Filename should be created from current date
+    const date = new Date().toISOString().split('T')[0];
```

Comment:
Hhmm, I wonder if we should create some utility function for this (probably exported from ./index.tsx and imported here).
Looks like we have this `new Date(n).toISOString().split("T")[0]` pattern in at least 3-4 places

```sh
git grep -n ".toISOString()" packages/fxa-settings | cat

packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.test.tsx:58:      .toISOString()
packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.test.tsx:65:    const date = new Date().toISOString().split('T')[0];
packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.test.tsx:83:    const date = currentDate.toISOString().split('T')[0];
packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.tsx:85:    const date = currentDate.toISOString().split('T')[0];
```

Not sure about these [from ./index.tsx], which seem to be shared between the component and tests:
```js
    const maxLength = 70;
    const prefix = 'Firefox-Recovery-Key';
```

Or a utility function for creating a filename from an email and optional date.